📝 Update setup document (#3091)
3scale/porta: fedora_manual_setup.md

> # Manual setup on Fedora (34)

Reviewer: ```suggestion
# Manual setup on Fedora (36)
```
I got Fedora 36 and this manual is valid

Author: I'm trying to replace all instructions with Podman but it's not as simplye as I thought. I will leave this change to someone who actually knows how Podman works, if that's OK.

Reviewer: I'll do it myself today. Don't merge this

## Language and runtime version management

In order to manage the required languages and runtime: Ruby, Python and Node.js, we recommend using [asdf](https://asdf-vm.com/guide/introduction.html). See its [getting started](https://asdf-vm.com/guide/getting-started.html) guide in order to get specific instructions for your SHELL and installation method.

Once installed, use the provided `.tool-versions.sample` file to get the appropriate versions.
```
ln -s .tool-versions.sample .tool-versions
```

## Ruby and Node.js

The project supports **[ruby 2.6.x](https://www.ruby-lang.org/en/downloads/)** and **[Node.js 12](https://nodejs.org/en/download/)**.
The recommended way to install them is with `asdf`:

```
asdf plugin add ruby
asdf plugin add nodejs

asdf install
```

> Alternatively, Node.js can be installed as a [Module](https://developer.fedoraproject.org/tech/languages/nodejs/nodejs.html):
> ```
> dnf module install nodejs:12
> ```

## Dependencies

```
sudo yum install patch autoconf automake bison libffi-devel libtool libyaml-devel readline-devel sqlite-devel zlib-devel openssl-devel ImageMagick ImageMagick-devel mysql-devel postgresql-devel chromedriver memcached

> sudo systemctl restart memcached

Reviewer: Not sure about installing memcached, b/c I use it in a container, and the APP will work even without it anyway

Author: I don't use it either. Actually, there is no step or instruction to start it up as part of macOS setup.

```

## Database (Postgres / MySQL / Oracle)

> Postgres, MySQL or Oracle has to be running for the application to work. The easiest way to do it is in a [Docker](https://www.docker.com/) container by simply running:

Reviewer: ```suggestion
Postgres, MySQL or Oracle has to be running for the application to work. The easiest way to do it is in a [Podman](https://podman.io/) container by simply running:
```
In all this document, I think we should replace "Docker" by "Podman" or "Container". To follow RH guidelines.

```
docker run -d -p 3306:3306 -e MYSQL_ALLOW_EMPTY_PASSWORD=true --name mysql57 mysql:5.7
```

Alternatively, you can run Postgres directly on your machine by following [this article](https://developer.fedoraproject.org/tech/database/postgresql/about.html).

## Sphinx Search

Install package and run it

```
sudo dnf install sphinx
bundle exec rake ts:configure ts:start
```

## Redis

[Redis](https://redis.io) has to be running for the application to work. The easiest way to do it is in a [Docker](https://www.docker.com/) container by simply running:

```
docker run -d -p 6379:6379 redis
```

Alternatively, you can run Redis directly on your machine by using `yum`:

```
sudo yum install redis

sudo systemctl restart redis
```

> 

Reviewer: ```suggestion

## Memcached

TODO: A brief explanation here...

docker run -d -p 11211:11211 memcached
```

Better recommend using docker? like any other dependency, If we don't recommend to install the database, which is required, and suggest to use docker instead, why not doing the same with memcached, which is optional?

Author: I for instance try to avoid using Docker due to memory consumption but I see your point. I will add this new section and maybe mention the 2 options. I need to do the same for the macOS document.

## Eventmachine

> Eventmachine has to be installed with `--with-cppflags=-I/usr/include/openssl/`. Simply run:

Reviewer: What's this for? I'm not sure I installed this.

Author: This probably fixed some issue in the past but nobody knew why. We can maybe trace the commit back and see what happened.

```
bundle config build.eventmachine --with-cppflags=-I/usr/include/openssl/
```

## Bundler
We manage our gems with [Bundler](https://bundler.io/). Install it by running:

```
gem install bundler
```

Then install all required Ruby gems:

```
bundle install
```

## Yarn (1.x)

To manage our JavaScript packages we use [Yarn](https://yarnpkg.com/). It is recommended to install it with NPM:
```
npm install --global yarn
```

> > See other ways to install it, such as with `brew`, at https://classic.yarnpkg.com/en/docs/install.

Reviewer: Why linking to the installer for versions lower than 2.0?

Author: Because we still use 1.x
